### DISCUSSION TITLE
How to clear row selection using an external button

### DISCUSSION THREAD

**Discussion by rajnishrao** [TanStack/table#3480] [Q&A]
Hi,

I am using table with `useRowSelect` checkbox. I have a button in a parent component that I am using to clear row selection. I used `forwardRef` to evoke `toggleAllRowsSelected()` function. However, instead of toggle, I just wanted to evoke this function as clear row selection when all or any count of row is selected. I see there `isAllRowsSelected` but not able to use this as a check. here is the example code https://codesandbox.io/s/forward-ref-57bwn?file=/src/components/Table.js

Thanks & Regards,
Rajnish

**Comment by ivan-sakoman** [comment]
you just need to pass "false" in the function like this: "toggleAllRowsSelected(false)"
